prometheus/value.go

```diff
@@ -82,15 +85,25 @@ func (v *value) SetToCurrentTime() {
 	v.Set(float64(time.Now().UnixNano()) / 1e9)
 }
 
+// add adjusts the underlying int64
+func (v *value) add(delta int64) {
+	atomic.AddInt64(&v.valInt, delta)
+}
+
 func (v *value) Inc() {
-	v.Add(1)
+	v.add(1)
 }
 
 func (v *value) Dec() {
-	v.Add(-1)
+	v.add(-1)
 }
 
 func (v *value) Add(val float64) {
+	if math.Trunc(val) == val {
+		atomic.AddInt64(&v.valInt, int64(val))
+		return
+	}
```

Comment:
This has edge cases that won't end well, e.g. +Inf or -Inf. Or just very large numbers that would overflow an int64 but would still work fine with float64 (at least approximately).

How about this:
```Go
ival := int64(val)
if float64(ival) == val {
	v.add(ival)
	return
}
```

It would also be good to have some of those edge cases in the tests (e.g. `Add(+Inf)` or `Add(1e99)`).